scripts/notarize_build.py

```diff
@@ -0,0 +1,224 @@
+from alws.utils.fastapi_sqla_setup import setup_all
+from alws.utils.pulp_client import PulpClient
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    level=logging.INFO,
+    datefmt="%Y-%m-%d %H:%M:%S",
+    handlers=[
+        logging.StreamHandler(),
+    ],
+)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        "build-notarizer",
+        description="Creates records in immudb for the further usage in generating SBOM data",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        '--build-id',
+        type=int,
+        help='Build ID to process',
+        required=True,
+    )
+    parser.add_argument(
+        '--pulp-host',
+        type=str,
+        help='Pulp host',
+        required=True,
+    )
+    parser.add_argument(
+        '--pulp-username',
+        type=str,
+        help='Pulp username',
+        required=True,
+    )
+    parser.add_argument(
+        '--pulp-password',
+        type=str,
+        help='Pulp password',
+        required=True,
+    )
+    parser.add_argument(
+        '--immudb-address',
+        type=str,
+        help='Immudb address',
+        required=True,
+    )
+    parser.add_argument(
+        '--immudb-username',
+        type=str,
+        help='Immudb username',
+        required=True,
+    )
+    parser.add_argument(
+        '--immudb-password',
+        type=str,
+        help='Immudb password',
+        required=True,
+    )
+    parser.add_argument(
+        '--immudb-database',
+        type=str,
+        help='Immudb database',
+        required=True,
+    )
+    return parser.parse_args()
+
+
+def extract_git_metadata(
+    task: models.BuildTask,
+    immudb_wrapper: ImmudbWrapper,
+) -> dict:
+    metadata = {
+        'source_type': 'git',
+        'git_url': task.ref.url,
+        'git_ref': task.ref.git_ref,
+        'git_commit': task.ref.git_commit_hash,
+    }
+    if not task.ref.git_commit_hash:
+        return metadata
+    with tempfile.TemporaryDirectory() as tmpdir:
+        subprocess.run(
+            args=('git', 'clone', task.ref.url, '.'),
+            cwd=tmpdir,
+            check=True,
+            capture_output=True,
+            encoding='utf-8',
+        )
+        subprocess.run(
+            args=('git', 'checkout', task.ref.git_commit_hash),
+            cwd=tmpdir,
+            check=True,
+            capture_output=True,
+            encoding='utf-8',
+        )
+        result = immudb_wrapper.authenticate_git_repo(tmpdir)
+        if result:
+            metadata['alma_commit_sbom_hash'] = (
+                result.get('value', {})
+                .get('Metadata', {})
+                .get('git', {})
+                .get('Commit')
+            )
+    return metadata
+
+
+async def main(
+    need_to_re_notarize: bool = False,
+):
+    args = parse_args()
+    pulp_client = PulpClient(
+        host=args.pulp_host,
```

Comment:
Just a suggestion. How if we try resolving some of the configuration values from env variables? I'm mostly thinking in those used by albs-web-server. This way we can run it from a web_server container which, if not wrong, will have the env variables in place and we won't need to look for credentials, copy/paste ...